lib/internal/readline/eventsToAsyncIteratorFactory.js

```diff
@@ -0,0 +1,107 @@
+'use strict';
+const {
+  Promise,
+  SymbolAsyncIterator,
+  ArrayPrototypeConcat,
+} = primordials;
+const FixedQueue = require('internal/fixed_queue');
+
+const PAUSE_THRESHOLD = 1024;
+const RESUME_THRESHOLD = 1;
+const ITEM_EVENTS = ['data'];
+const CLOSE_EVENTS = ['close', 'end'];
+const ERROR_EVENTS = ['error'];
+
+
+function waitNext(emitter, next, events) {
+  return new Promise((resolve, reject) => {
+    const resolveNext = () => {
+      for (let i = 0; i < events.length; i++)
+        emitter.off(events[i], resolveNext);
+      try {
+        resolve(next());
+      } catch (promiseError) {
+        reject(promiseError);
+      }
+    };
+    for (let i = 0; i < events.length; i++)
+      emitter.on(events[i], resolveNext);
+  });
+}
+
+module.exports = function eventsToAsyncIteratorFactory(readable, {
+  pauseThreshold = PAUSE_THRESHOLD,
```

Comment:
I am not sure this is how we do constants in terms of naming but I've found several inconsistencies and our style guide doesn't [mention it](https://github.com/nodejs/node/blob/master/doc/README.md) from what I can tell. Odd.

cc @mcollina is this the convention for constants?

Reply:
Yeah, I was taking a look too and the constant naming varies from file to file.
Once you tell me the preferable style I'm happy to fix it, if necessary.
![image](https://user-images.githubusercontent.com/5713887/147661069-ecf7d7ce-91fe-4c8e-ab40-300aa83907d6.png)

I also looked up on the md files  I didn't find any rules about it

Reply:
Yeah, I was taking a look and the styles vary from file to file, and I couldn't find any rule on the mds.

Once you tell me the preferable style I'm happy to fix it, if necessary

Reply:
General rule of thumb I follow is `ALL_CAPS_SNAKE` for user-facing constants, and `kCamelCase` for internal only constants.

Reply:
That sounds good. I can follow the same rule here